rfcs/button-group-component.md

```diff
@@ -0,0 +1,77 @@
+# Button Group Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button} from 'baseui/button';
+import {ButtonGroup} from 'baseui/button-group';
+
+export default () => {
+  return
+    <ButtonGroup>
+      <Button>Copy</Button>
+      <Button>Cut</Button>
+      <Button>Paste</Button>
+    </ButtonGroup>
+}
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {Button} from 'baseui/button';
+import {ButtonGroup} from 'baseui/button-group';
+
+export default () => {
+  return
+    <ButtonGroup
+      defaultChecked={[0, 1]}
+      mode="checkbox">
+      <Button>Bold</Button>
+      <Button>Italic</Button>
+      <Button>Underline</Button>
+    </ButtonGroup>
+}
+```
+
+## Exports
+
+* `ButtonGroup`
+* `StyledRoot`
+
+## `ButtonGroup` API
+
+* `children: (props: Props) => React.Node` - Required
+  This can be an SVG icon as well.
+* `ariaLabel: string` - Optional:
+  Accessible label
+* `mode: string` - Optional:
+  Use the `mode` prop to render toggleable Buttons:
+  * the value `radio` will cause Buttons to behave like radio buttons,
+  * the value `checkbox` will cause Buttons to behave like checkboxes.
+* `defaultChecked: number | Array<number>`
+  Index or array of indices of the selected Button(s).
+  Primarily for use with uncontrolled components with a `mode` prop defined.
+* `checked: number | Array<number>` - Optional
+  Index or array of indices of the selected Button(s).
+  Primarily for use with controlled components with a `mode` prop defined.
```

Comment:
So the `defaultChecked`/`checked` props sort of allow you to use this component as both a controlled (stateless) and uncontrolled (stateful) component. If we want to be more consistent with the rest of our BaseUI components, it may be preferable to have a `StatefulButtonGroup` with an `initialState prop` and a `ButtonGroup` with the `checked` prop.

Reply:
ah, you are right, I'll update it